parlai/mturk/core/mturk_manager.py

```diff
@@ -151,16 +151,18 @@ def _reset_time_logs(self, force=False):
         with LockFile(file_lock) as _lock_file:
             assert _lock_file is not None
             if os.path.exists(file_path):
-                if not force:
-                    with open(file_path, 'rb+') as time_log_file:
-                        existing_times = pickle.load(time_log_file)
-                        if time.time() - existing_times['last_reset'] > \
-                                24 * 60 * 60:  # Reset if more than a day old
-                            pass
-                        elif time.time() - existing_times['last_reset'] < \
-                                RESET_TIME_LOG_TIMEOUT:
-                            # no need to reset, another thread did recently
-                            return
+                with open(file_path, 'rb+') as time_log_file:
+                    existing_times = pickle.load(time_log_file)
+                    if time.time() - existing_times['last_reset'] > \
```

Comment:
sorry, maybe i am confused - so this checks if the last reset was greater than a day, and then does nothing if so?

Reply:
If the time has been more than a day since the last reset as listed by the file, then it passes out of this conditional and resets the file. 
If the file has been reset really recently, it returns and doesn't reset the logs.

This logic is indeed opaque, I'll update